### PR TITLE
OSDOCS-4568 - RNs Compliance Operator 0.1.59

### PR DESCRIPTION
**Pull request by Wiharris**
**[OSDOCS-4568](https://issues.redhat.com//browse/OSDOCS-4568)**
Adds Release Notes for Compliance Operator 0.1.59

**Version(s):**

- PR applies to 4.8+

**Issue:**
https://issues.redhat.com/browse/OSDOCS-4568

**Link to docs preview:**

- [OpenShift Compliance Operator 0.1.59](https://52987--docspreview.netlify.app/openshift-enterprise/latest/security/compliance_operator/compliance-operator-release-notes.html#compliance-operator-release-notes-0-1-59)
- [Compliance profiles](https://52987--docspreview.netlify.app/openshift-enterprise/latest/security/compliance_operator/compliance-operator-supported-profiles.html#compliance-supported-profiles_compliance-operator-supported-profiles)

**Additional information:**
Links to advisory will not show until Errata is `shipped live`

### CODE REVIEW

File: modules/compliance-supported-profiles.adoc [hunk 1]
```diff
@@ -86,13 +86,13 @@ The Compliance Operator provides the following compliance profiles:
 |0.1.47+
 |link:https://www.pcisecuritystandards.org/document_library?document=pci_dss[PCI Security Standards &#174; Council Document Library]
 |`x86_64`
-
+ `ppc64le`
```
Comment (sheriff-rh) [inline modules/compliance-supported-profiles.adoc:89]: I know this still renders correctly, but include an empty line below `ppc64le`  so that table entries do not run together.

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 2]
```diff
@@ -13,6 +13,24 @@ These release notes track the development of the Compliance Operator in the {pro
...
+The following advisory is available for the OpenShift Compliance Operator 0.1.59:
+
+* link:https://access.redhat.com/errata/RHBA-2022:8538[RHBA-2022:8538 - OpenShift Compliance Operator bug fix update]
+
+[id="compliance-operator-0-1-59-bug-fixes"]
+=== Bug fixes
+
+* Previously, support for Payment Card Industry Data Security Standard (PCI DSS) and PCI DSS node did not show on `ppc64le`, which left profiles disabled in the architecture. Now, support for PCI DSS and PCI DSS node shows on `ppc64le` and enables all profiles in the architecture. (link:https://issues.redhat.com/browse/OCPBUGS-3252[*OCPBUGS-3252*])
+
+* Previously, after the recent upgrade to version 0.1.57 , the `rerunner` service account (SA) was no longer owned by the cluster service version (CSV), which caused the SA to be removed. Now, the CSV owns the `rerunner` SA in 0.1.57. (link:https://issues.redhat.com/browse/OCPBUGS-3452[*OCPBUGS-3452*])
```
Comment (sheriff-rh) [inline security/compliance_operator/compliance-operator-release-notes.adoc:28]: version 0.1.57, (no space)

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 3]
```diff
@@ -13,6 +13,24 @@ These release notes track the development of the Compliance Operator in the {pro
...
+
+* Previously, after the recent upgrade to version 0.1.57, the metrics configuration default was 8080 and the `TargetDown` alarm was removed from the compliance operator. Now, the metrics configuration default is 8383 and the `TargetDown` remains in the compliance operator after 0.1.57. (link:https://issues.redhat.com/browse/OCPBUGS-3097[*OCPBUGS-3097*])
```
Comment (sheriff-rh) [inline security/compliance_operator/compliance-operator-release-notes.adoc:30]: Let's put backticks around the port numbers and also state that they are indeed port numbers for clarity:

was port `8080` 

... default is port `8383` 

... Compliance Operator after 0.1.57. (Always capitalize Compliance Operator, File Integrity Operator, etc)

File: modules/compliance-supported-profiles.adoc (hunk 1 above)
Comment (sheriff-rh) [reply modules/compliance-supported-profiles.adoc:89]: @prb112 just want to confirm with you this is okay to add back in now.

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 4]
```diff
@@ -13,6 +13,24 @@ These release notes track the development of the Compliance Operator in the {pro
...
+== OpenShift Compliance Operator 0.1.59
+
+The following advisory is available for the OpenShift Compliance Operator 0.1.59:
+
+* link:https://access.redhat.com/errata/RHBA-2022:8538[RHBA-2022:8538 - OpenShift Compliance Operator bug fix update]
+
+[id="compliance-operator-0-1-59-bug-fixes"]
+=== Bug fixes
+
+* Previously, support for Payment Card Industry Data Security Standard (PCI DSS) and PCI DSS node did not show on `ppc64le`, which left profiles disabled in the architecture. Now, support for PCI DSS and PCI DSS node shows on `ppc64le` and enables all profiles in the architecture. (link:https://issues.redhat.com/browse/OCPBUGS-3252[*OCPBUGS-3252*])
```
Comment (sheriff-rh) [inline security/compliance_operator/compliance-operator-release-notes.adoc:26]: I like that you defined PCI  DSS, let's keep that! 

We are referring specifically to `ocp4-pci-dss` and `ocp4-pci-dss-node` compliance profiles, so maybe something like: 
"... (PCI DSS) compliance profiles `ocp4-pci-dss` and `ocp4-pci-dss-node` did not show..."

Comment (sheriff-rh) [reply security/compliance_operator/compliance-operator-release-notes.adoc:26]: Additionally, I think we need to change some language here. "Did not show" is ambiguous. Were they unavailable? Disabled? 

Is this a new feature? I don't see PCI DSS support in 0.1.53 and prior releases ([4.6 and 0.1.53 latest release](https://docs.openshift.com/container-platform/4.6/security/compliance_operator/compliance-operator-supported-profiles.html), for example)

**Review by sheriff-rh — CHANGES_REQUESTED** [review]
Looks really good! I left a few comments to provide some context and clarity.

File: security/compliance_operator/compliance-operator-release-notes.adoc (hunk 3 above)
Comment (Wiharris) [reply security/compliance_operator/compliance-operator-release-notes.adoc]: Updated to reflect suggested changes! :)

File: security/compliance_operator/compliance-operator-release-notes.adoc (hunk 4 above)
Comment (Wiharris) [reply security/compliance_operator/compliance-operator-release-notes.adoc]: Added PCI DSS profile as an enhancement since it's new functionality for different architectures.

File: modules/compliance-supported-profiles.adoc (hunk 1 above)
Comment (prb112) [reply modules/compliance-supported-profiles.adoc:89]: it's OK to add back ppc64le. Thank you

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 5]
```diff
@@ -13,6 +13,30 @@ These release notes track the development of the Compliance Operator in the {pro
...
+
+The following advisory is available for the OpenShift Compliance Operator 0.1.59:
+
+* link:https://access.redhat.com/errata/RHBA-2022:8538[RHBA-2022:8538 - OpenShift Compliance Operator bug fix update]
+
+
+[id="compliance-operator-0-1-59-new-features-and-enhancements"]
+=== New features and enhancements
+
+The Compliance Operator now supports Payment Card Industry Data Security Standard (PCI DSS) `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on all architectures.
```
Comment (rhmdnd) [inline security/compliance_operator/compliance-operator-release-notes.adoc:27]: We could clarify this by saying specifically `ppc64le` and `s390x`, which are the architectures that were recently added for PCI-DSS.

This will provide more clarity for the reader since we've supported PCI-DSS on `x86_64` for a while.

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 6]
```diff
@@ -13,6 +13,30 @@ These release notes track the development of the Compliance Operator in the {pro
...
+
+[id="compliance-operator-0-1-59-bug-fixes"]
+=== Bug fixes
+
+* Previously, the Compliance Operator did not support the Payment Card Industry Data Security Standard (PCI DSS) `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on different architectures such as `ppc64le`. Now, the Compliance Operator supports `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on `ppc64le` and all architectures. (link:https://issues.redhat.com/browse/OCPBUGS-3252[*OCPBUGS-3252*])
```
Comment (rhmdnd) [inline security/compliance_operator/compliance-operator-release-notes.adoc:32]: Similar comment here as above.

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 7]
```diff
@@ -13,6 +13,30 @@ These release notes track the development of the Compliance Operator in the {pro
...
+
+* Previously, after the recent upgrade to version 0.1.57, the `rerunner` service account (SA) was no longer owned by the cluster service version (CSV), which caused the SA to be removed. Now, the CSV owns the `rerunner` SA in 0.1.57. (link:https://issues.redhat.com/browse/OCPBUGS-3452[*OCPBUGS-3452*])
+
+* Previously, after the recent upgrade to version 0.1.57, the metrics configuration default was `8080` and the `TargetDown` alarm was removed from the Compliance Operator. Now, the metrics configuration default is `8383` and the `TargetDown` remains in the Compliance Operator after 0.1.57. (link:https://issues.redhat.com/browse/OCPBUGS-3097[*OCPBUGS-3097*])
```
Comment (rhmdnd) [inline security/compliance_operator/compliance-operator-release-notes.adoc:36]: I might suggest rewording this to be a cause and effect.

With CO 0.1.57, the default metrics port changed but CO wasn't using the new default. This resulted in `TargetDown` alerts because the metrics service couldn't connect to the compliance operator's metrics endpoint.

With 0.1.59, we've updated the compliance operator to use the correct default for metrics.

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 8]
```diff
@@ -13,6 +13,30 @@ These release notes track the development of the Compliance Operator in the {pro
...
+
+* Previously, {product-title} version detection did not work on IBM cloud nor HyperShift, which left the application programming interface (API) path unspecified. This fix introduces variables that specify the API path, ensuring {product-title} version detection works on IBM cloud and HyperShift. (link:https://issues.redhat.com/browse/CMP-1587[*CMP-1587*])
```
Comment (jhrozek) [inline security/compliance_operator/compliance-operator-release-notes.adoc:38]: Unfortunately we should back off this fix from the downstream release notes -- the fix does not work as adverised (see the last comment in https://issues.redhat.com/browse/OCPBUGS-3953)

File: security/compliance_operator/compliance-operator-release-notes.adoc (hunk 5 above)
Comment (Wiharris) [reply security/compliance_operator/compliance-operator-release-notes.adoc]: Updated! Removed s390x since  s390x is exclude from PCI-DSS architectures

File: security/compliance_operator/compliance-operator-release-notes.adoc (hunk 8 above)
Comment (Wiharris) [reply security/compliance_operator/compliance-operator-release-notes.adoc]: @jhrozek I have removed this fix from the PR.

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 9]
```diff
@@ -13,6 +13,31 @@ These release notes track the development of the Compliance Operator in the {pro
...
+
+* link:https://access.redhat.com/errata/RHBA-2022:8538[RHBA-2022:8538 - OpenShift Compliance Operator bug fix update]
+
+
+[id="compliance-operator-0-1-59-new-features-and-enhancements"]
+=== New features and enhancements
+
+The Compliance Operator now supports Payment Card Industry Data Security Standard (PCI DSS) `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on the following architectures:
+
+* `x86_64`
```
Comment (rhmdnd) [inline security/compliance_operator/compliance-operator-release-notes.adoc:29]: We've technically supported PCI-DSS on `x86_64` since 0.1.47. We could shorten this note up to only say that `ppc64le` is supported.

I'm wondering if people would find it confusing if we keep `x86_64` in the note, or if they think it wasn't supported before.

https://docs.openshift.com/container-platform/4.11/security/compliance_operator/compliance-operator-release-notes.html#compliance-operator-release-notes-0-1-47

Comment (Wiharris) [reply security/compliance_operator/compliance-operator-release-notes.adoc]: Done! :)

**Review by rhmdnd — COMMENTED** [review]
Looks good to me. Thanks for the updates.

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 10]
```diff
@@ -13,6 +13,28 @@ These release notes track the development of the Compliance Operator in the {pro
...
+
+[id="compliance-operator-0-1-59-bug-fixes"]
+=== Bug fixes
+
+* Previously, the Compliance Operator did not support the Payment Card Industry Data Security Standard (PCI DSS) `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on different architectures such as `ppc64le`. Now, the Compliance Operator supports `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on `ppc64le` architecture. (link:https://issues.redhat.com/browse/OCPBUGS-3252[*OCPBUGS-3252*])
+
+* Previously, after the recent upgrade to version 0.1.57, the `rerunner` service account (SA) was no longer owned by the cluster service version (CSV), which caused the SA to be removed. Now, the CSV owns the `rerunner` SA in 0.1.57. (link:https://issues.redhat.com/browse/OCPBUGS-3452[*OCPBUGS-3452*])
+
+* In 0.1.57, the default metrics port was upgraded; however, the Compliance Operator did not use the new default metrics port. This resulted in `TargetDown` alerts due to the metrics service not connecting to the Compliance Operator's metrics endpoint. With 0.1.59, the Compliance Operator now uses the new default metrics port. (link:https://issues.redhat.com/browse/OCPBUGS-3097[*OCPBUGS-3097*])
```
Comment (mrogers950) [inline security/compliance_operator/compliance-operator-release-notes.adoc:36]: I think this could be clarified some if we mention the port number. I.e.,

In 0.1.57, the operator started the controller metrics endpoint listening on port 8080. This resulted in `TargetDown` alerts since cluster monitoring expected port 8383.  With 0.1.59, the operator starts the endpoint listening on port 8383 as expected.

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 11]
```diff
@@ -13,6 +13,28 @@ These release notes track the development of the Compliance Operator in the {pro
...
+=== New features and enhancements
+
+The Compliance Operator now supports Payment Card Industry Data Security Standard (PCI-DSS) `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on `ppc64le` alongside `x86_64` architectures. 
+
+[id="compliance-operator-0-1-59-bug-fixes"]
+=== Bug fixes
+
+* Previously, the Compliance Operator did not support the Payment Card Industry Data Security Standard (PCI DSS) `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on different architectures such as `ppc64le`. Now, the Compliance Operator supports `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on `ppc64le` architecture. (link:https://issues.redhat.com/browse/OCPBUGS-3252[*OCPBUGS-3252*])
+
+* Previously, after the recent upgrade to version 0.1.57, the `rerunner` service account (SA) was no longer owned by the cluster service version (CSV), which caused the SA to be removed. Now, the CSV owns the `rerunner` SA in 0.1.57. (link:https://issues.redhat.com/browse/OCPBUGS-3452[*OCPBUGS-3452*])
```
Comment (mrogers950) [inline security/compliance_operator/compliance-operator-release-notes.adoc:34]: Suggestion for part of this (also, I think the second reference to 0.1.57 should be 0.1.59)
```
... which caused the SA to be removed during the operator upgrade. Now, the CSV owns the `rerunner` SA in 0.1.59, and an upgrade from any previous version will not result in a missing SA.
```

**Review by GroceryBoyJr — COMMENTED** [review]
@Wiharris I'm standing down from peer reviewing, you are getting tech review at this time.

File: security/compliance_operator/compliance-operator-release-notes.adoc (hunk 10 above)
Comment (Wiharris) [reply security/compliance_operator/compliance-operator-release-notes.adoc]: Updated! :)

File: security/compliance_operator/compliance-operator-release-notes.adoc [hunk 12]
```diff
@@ -13,6 +13,28 @@ These release notes track the development of the Compliance Operator in the {pro
...
+
+The following advisory is available for the OpenShift Compliance Operator 0.1.59:
+
+* link:https://access.redhat.com/errata/RHBA-2022:8538[RHBA-2022:8538 - OpenShift Compliance Operator bug fix update]
+
+
+[id="compliance-operator-0-1-59-new-features-and-enhancements"]
+=== New features and enhancements
+
+The Compliance Operator now supports Payment Card Industry Data Security Standard (PCI-DSS) `ocp4-pci-dss` and `ocp4-pci-dss-node` profiles on `ppc64le` alongside `x86_64` architectures.
```
[the remark this answers is not in the document]
Comment (Wiharris) [reply security/compliance_operator/compliance-operator-release-notes.adoc]: I have removed x86_64!

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM

**Comment by mburke5678** [comment]
@Wiharris Did this PR get a QE review?

**Comment by Wiharris** [comment]
> @Wiharris Did this PR get a QE review?

@mburke5678 Usually we do not require QE for RNs. It has been tech reviewed, peer, and by engineer.
